CASSGO-33 snappy has been moved to a separate package

OleksiienkoMykyta opened the pull request:
This PR should be reviewed right after https://github.com/apache/cassandra-gocql-driver/pull/1822 is merged.

joao-r-reis commented:
@OleksiienkoMykyta can you rebase this branch? Or even start from scratch (from trunk) and force push to this branch, might be easier

OleksiienkoMykyta commented:
> @OleksiienkoMykyta can you rebase this branch? Or even start from scratch (from trunk) and force push to this branch, might be easier

Sure, I'll handle this on the next week

OleksiienkoMykyta committed:
Move the snappy compressor into a separate go package

Currently, Snappy is downloaded and included even when users only require LZ4 compression.
To streamline the driver and reduce unnecessary dependency overhead for users who don't utilize Snappy,
move the Snappy compressor into its own separate package.
This will allow users to include only the compression dependencies they need.

Patch by Mykyta Oleksiienko; reviewed by Joao Reis CASSGO-33

OleksiienkoMykyta commented:
@joao-r-reis, I have fixed the conflicts.

joao-r-reis commented on CHANGELOG.md:
```diff
@@ -18,6 +18,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
+- Moved the Snappy compressor into its own separate package.
+  This allows users to include only the compression dependencies they need. (CASSGO-33)
```
I think we don't need this last line, it would be good to keep this list brief, users can open the linked JIRAs if they want more details.

OleksiienkoMykyta replied:
I agree, fixed.

joao-r-reis commented on snappy/snappy-compressor.go:
```diff
@@ -0,0 +1,28 @@
+package snappy
+
+import "github.com/golang/snappy"
```
We can have this file be named `compressor.go` no?

OleksiienkoMykyta replied:
I tried to align it with lz4, but sure, we can keep it just as a `compressor.go`.
Fixed.

joao-r-reis reviewed (commented):
Left 1 comment about the change in CHANGELOG.md and the new file name for the snappy compressor.

I'm not sure if the tests under `snappy/compressor_test.go` will run. I believe this is being handled in https://github.com/apache/cassandra-gocql-driver/pull/1864 but it would be nice to get confirmation that these tests are passing before merging this.

Maybe we can wait for #1864 to be merged before merging this one.

OleksiienkoMykyta commented:
> Left 1 comment about the change in CHANGELOG.md and the new file name for the snappy compressor.
> 
> I'm not sure if the tests under `snappy/compressor_test.go` will run. I believe this is being handled in #1864 but it would be nice to get confirmation that these tests are passing before merging this.
> 
> Maybe we can wait for #1864 to be merged before merging this one.

Sure, let's wait for it. It's always better to double-check

joao-r-reis commented:
Can you rebase this branch so it gets the changes from trunk? With this I think we will see the snappy unit tests run on CI

joao-r-reis commented on snappy/compressor_test.go:
```diff
@@ -22,7 +22,7 @@
  */
 
-package gocql
+package snappy
```
This file is missing 
``` 
//go:build all || unit
// +build all unit
```
so it runs in the CI unit tests step

joao-r-reis commented:
integration tests with snappy appear fine so when you [add this](https://github.com/apache/cassandra-gocql-driver/pull/1844#discussion_r2054419043) I'll merge this pr

joao-r-reis commented:
@OleksiienkoMykyta do you have time to address [this](https://github.com/apache/cassandra-gocql-driver/pull/1844#issuecomment-2821807834)? We're trying to resolve the remaining pending PRs for the 2.0 release
